test/irmin-pack/test_pack_version_bump.ml

```diff
@@ -0,0 +1,169 @@
+(** These tests for issue #1658, which follows PR #1655 which
+   introduces a store version bump from V1 to V2.
+
+There are 3 tests:
+
+- test_RO_no_version_bump: open a V1 store RO mode, the version should
+  remain V1
+
+- test_RW_no_version_bump: open a V1 store RW mode, but don't attempt
+  to mutate, version should remain V1
+
+- test_RW_version_bump: open a V1 store RW mode, change something,
+  version should bump to V2
+
+*)
+
+
+open! Import
+open Common
+
+let src = Logs.Src.create "tests.version_bump" ~doc:"Test pack version bump"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+module Util = struct
+
+  (** Following are generic utils *)
+
+  let tmp_dir () = Filename.temp_file "test_pack_version_bump_" ""
+
+  (** Make a directory; dir is assumed to not exist; parent is assumed to exist *)
+  let mkdir p =
+    assert(not (Sys.file_exists p));
+    Filename.quote_command "mkdir" [p] |> fun cmd ->
+    [%log.info "exec: %s\n%!" cmd];
+    Sys.command cmd |> function
+    | 0 -> ()
+    | n ->
+      Fmt.failwith "Command `%s' exited with non-zero code %d\n" cmd n
+
+
+  (** Copy src to dst; dst is assumed to not exist *)
+  let copy_dir src dst =
+    assert(not (Sys.file_exists dst));
+    (* don't check if it is empty; perhaps we should *)
+    Filename.quote_command "cp" ["-R";src;dst] |> fun cmd ->
+    [%log.info "exec: %s\n%!" cmd];
+    Sys.command cmd |> function
+    | 0 -> ()
+    | n ->
+      Fmt.failwith "Failed to set up the test env; command `%s' exited \
+                    with non-zero code %d\n" cmd n
+
+  (** More specific utils from here *)
+
+  let v1_store_archive_dir = "test/irmin-pack/data/version_1"
+
+  (** Check we are in the location we expect, the project root. See
+     test_existing_stores.goto_project_root for alternative. *)
+  let assert_project_root () =
+    match Sys.file_exists v1_store_archive_dir with
+    | true -> ()
+    | false ->
+      Fmt.failwith "Couldn't find dir %S from cwd %S\n%!"
+        v1_store_archive_dir
+        Filename.current_dir_name
+
+  let _ = assert_project_root ()
+
+  module Unix_ = Irmin_pack.IO.Unix
+
+  (** Get the version of the underlying file; file is assumed to
+     exist; file is assumed to be an Irmin_pack.IO.Unix file *)
+  let io_get_version ~fn : [`V1 | `V2] =
+    assert(Sys.file_exists fn);
```

Comment:
This assertion should be implied by `~fresh:false`?

Reply:
Probably, but adding the assertion explicitly makes it clearer that this is required, I think, and there isn't much downside.